packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -378,13 +379,30 @@ gmaps.InfoWindowOptions _infoWindowOptionsFromMarker(Marker marker) {
     return null;
   }
 
-  final content = '<h3 class="infowindow-title">' +
-      sanitizeHtml(marker.infoWindow.title ?? "") +
-      '</h3>' +
-      sanitizeHtml(marker.infoWindow.snippet ?? "");
+  // Add an outer wrapper to the contents of the infowindow, we need it to listen
+  // to click events...
+  final container = DivElement()
+    ..id = 'gmaps-marker-${marker.markerId.value}-infowindow';
+  if (marker.infoWindow.title?.isNotEmpty ?? false) {
+    final title = HeadingElement.h3()
+      ..className = 'infowindow-title'
+      ..innerText = marker.infoWindow.title;
+    container.children.add(title);
+  }
+  if (marker.infoWindow.snippet?.isNotEmpty ?? false) {
+    final snippet = DivElement()
+      ..className = 'infowindow-snippet'
+      ..setInnerHtml(
+        sanitizeHtml(marker.infoWindow.snippet),
+        treeSanitizer: NodeTreeSanitizer.trusted,
+      );
+    container.children.add(snippet);
+  }
+
+  assert(container.children.isNotEmpty);
```

Comment:
is this case possible outside of the test environment?

Reply:
Nope, I'll remove the assert.